### DISCUSSION TITLE
Printing intermediate outputs in the compilation pipeline of Koka Source code to C

### DISCUSSION THREAD

**Discussion by swarnpriya** [koka-lang/koka#514] [General]
Hi, 
I wanted to know if we can see each intermediate representation of the compilation process. Like how using clang we can see the output after preprocessing, dump-ast, generate llvm-IR etc. What would be the equivalent steps in the Koka compilation?

**Comment by TimWhiting** [comment]
Passing `--core` will output a 'core' file. Right now, it just shows the final core (after optimizations, but before reference counting) I believe. There is also a `--showcore` which will print it to the console instead of to a file.

**Reply by swarnpriya** [reply]
Unfortunately, --showcore or --core doesn't seem to produce anything. Apart from these, I guess rest of the Koka options work.

**Reply by TimWhiting** [reply]
I've used `--core` recently and it has worked. Take a look in the output folder (usually .koka/../module_path.kkc). `.kkc` is the extension for core files, `.kki` for interface files.
